paillier_test.go

```diff
@@ -124,65 +124,85 @@ func TestCheckPlaintextSpace(t *testing.T) {
+	sk := CreateSecretKey(p, q)
 
-	cypher1, _ := privateKey.Encrypt(big.NewInt(5), rand.Reader)
-	cypher2, _ := privateKey.Encrypt(big.NewInt(6), rand.Reader)
-	cypher3, _ := privateKey.Encrypt(big.NewInt(7), rand.Reader)
-	cypher4, _ := privateKey.Encrypt(big.NewInt(8), rand.Reader)
-	cypher5 := privateKey.Add(cypher1, cypher2, cypher3, cypher4)
+	ct1, _ := sk.Encrypt(big.NewInt(5), rand.Reader)
+	ct2, _ := sk.Encrypt(big.NewInt(6), rand.Reader)
+	ct3, _ := sk.Encrypt(big.NewInt(7), rand.Reader)
+	ct4, _ := sk.Encrypt(big.NewInt(8), rand.Reader)
+	ct5 := sk.Add(ct1, ct2, ct3, ct4)
 
-	m := privateKey.Decrypt(cypher5)
+	m := sk.Decrypt(ct5)
 	if m.Cmp(big.NewInt(26)) != 0 {
 		t.Errorf("Unexpected decrypted value [%v]", m)
 	}
 }
 
-func TestAddCypherWithSmallKeyModulus(t *testing.T) {
-	privateKey := CreatePrivateKey(big.NewInt(7), big.NewInt(5))
+func TestAddCiphertextWithSmallKeyModulus(t *testing.T) {
+	sk := CreateSecretKey(big.NewInt(7), big.NewInt(5))
 
-	cypher1, _ := privateKey.Encrypt(big.NewInt(30), rand.Reader)
-	cypher2, _ := privateKey.Encrypt(big.NewInt(25), rand.Reader)
-	cypher3, _ := privateKey.Encrypt(big.NewInt(11), rand.Reader)
-	cypher4 := privateKey.Add(cypher1, cypher2, cypher3)
+	ct1, _ := sk.Encrypt(big.NewInt(30), rand.Reader)
+	ct2, _ := sk.Encrypt(big.NewInt(25), rand.Reader)
+	ct3, _ := sk.Encrypt(big.NewInt(11), rand.Reader)
+	ct4 := sk.Add(ct1, ct2, ct3)
 
-	m := privateKey.Decrypt(cypher4)
+	m := sk.Decrypt(ct4)
 	if m.Cmp(big.NewInt(31)) != 0 {
 		t.Errorf("Unexpected decrypted value [%v]", m)
 	}
 }
 
-func TestMulCypher(t *testing.T) {
-	privateKey := CreatePrivateKey(big.NewInt(17), big.NewInt(13))
+func TestMulCiphertext(t *testing.T) {
+	p, q := GenKeyPrimes(10)
+	sk := CreateSecretKey(p, q)
 
-	cypher, err := privateKey.Encrypt(big.NewInt(3), rand.Reader)
+	ct, err := sk.Encrypt(big.NewInt(3), rand.Reader)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	cypherMultiple := privateKey.Mul(cypher, big.NewInt(7))
-	multiple := privateKey.Decrypt(cypherMultiple)
+	ctMultiple := sk.Mul(ct, big.NewInt(7))
+	multiple := sk.Decrypt(ctMultiple)
 
 	// 3 * 7 = 21
 	if multiple.Cmp(big.NewInt(21)) != 0 {
 		t.Errorf("Unexpected decrypted value [%v]", multiple)
 	}
 }
 
-func TestMulCypherWithSmallKeyModulus(t *testing.T) {
-	privateKey := CreatePrivateKey(big.NewInt(7), big.NewInt(5))
+func TestMulCiphertextWithSmallKeyModulus(t *testing.T) {
+	sk := CreateSecretKey(big.NewInt(7), big.NewInt(5))
 
-	cypher, err := privateKey.Encrypt(big.NewInt(30), rand.Reader)
+	ct, err := sk.Encrypt(big.NewInt(30), rand.Reader)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	cypherMultiple := privateKey.Mul(cypher, big.NewInt(93))
-	multiple := privateKey.Decrypt(cypherMultiple)
+	ctMultiple := sk.Mul(ct, big.NewInt(93))
+	multiple := sk.Decrypt(ctMultiple)
 
 	// (30*93) mod (7*5) = 25
 	if multiple.Cmp(big.NewInt(25)) != 0 {
 		t.Errorf("Unexpected decrypted value [%v]", multiple)
 	}
 }
+
+func GenKeyPrimes(bits int) (p, q *big.Int) {
```

Comment:
Absolutely agree with @nkuba here. I understand the intention was to make those tests more real-world, but for the real-world Paillier, we'll probably use 1024 or 2048 modulus and this would have a significant impact on tests performance.